include/roboteam_ai/control/positionControl/PathPointNode.h

```diff
@@ -0,0 +1,40 @@
+//
+// Created by ratoone on 20-02-20.
+//
+
+#ifndef RTT_PATHPOINTNODE_H
+#define RTT_PATHPOINTNODE_H
+
+#include <roboteam_utils/Vector2.h>
+
+namespace rtt::ai::control{
+/**
+ * A node in the path tree generation. Has a position and a parent
+ */
+class PathPointNode {
+private:
+    Vector2 position;
+    PathPointNode *parent = nullptr;
```

Comment:
Why raw pointer?

Reply:
It can't be the object itself, as for some reason you can't declare a member of the same class. It can't be a reference, as it should be able to be null